### PR TITLE
fix issue 1998

### PR DESCRIPTION
**Pull request by xuezhaokun**
fix issue #1998

### CODE REVIEW

File: hs_core/search_indexes.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class BaseResourceIndex(indexes.SearchIndex, indexes.Indexable):
     public = indexes.BooleanField(faceted=True)
     discoverable = indexes.BooleanField(faceted=True)
     published = indexes.BooleanField(faceted=True)
+    is_replaced_by = indexes.BooleanField()
```
Comment (pkdash) [inline hs_core/search_indexes.py]: is this going to be facet?

Comment (pkdash) [inline hs_core/search_indexes.py]: Ignore the above comment.

File: hs_core/search_indexes.py [hunk 2]
```diff
@@ -159,6 +160,15 @@ def prepare_published(self, obj):
         else:
             return False
 
+    def prepare_is_replaced_by(self, obj):
+        if hasattr(obj, 'metadata'):
+            if obj.metadata.relations.all().filter(type='isReplacedBy').exists():
```
Comment (pkdash) [inline hs_core/search_indexes.py]: You can also do this: 
```return obj.metadata.relations.all().filter(type='isReplacedBy').exists()```

**Comment by pkdash** [comment]
@xuezhaokun Have left one minor comment. Otherwise, +1. Merge after Jenkins checks are successful.

**Comment by pkdash** [comment]
@xuezhaokun What's the status of this PR?

**Comment by xuezhaokun** [comment]
@pkdash I just discussed about this PR with @alvacouch yesterday. We noticed this issue is not that easy to solve. Because the new version could be private, but the old version is public. At this situation, we still need to make the old version be discoverable not the new version. We need to walk through all versions for the given resource and make the latest public one be discoverable. I need to make some modifications to this PR. Hopefully, I could find an easy solution for this issue as soon as possible.

**Comment by hyi** [comment]
@xuezhaokun @alvacouch @dtarb @horsburgh I believe the decision is to make the old version non discoverable. In fact, once a new version is created, the old version is obsolete, and will not show up in my resources page or anywhere, and can only be recovered when all its newer versioned counterparts are all deleted. I think you can remove all old versioned resources, i.e., with a isReplacedBy metadata relations, and make the new versioned resource discoverable only when it is made public although initially the new versioned resource is private by default.

**Comment by pkdash** [comment]
@xuezhaokun If it is simple fix then we can still consider this feature to be part release 1.10, otherwise, we may have to schedule it for next release.

**Comment by xuezhaokun** [comment]
@pkdash waiting for answers for Hong's comment. If Hong's comment works, it would be a really easy fix.

**Comment by alvacouch** [comment]
@hyi @pkdash this is a very simple fix that's guaranteed to work. It doesn't do some desirable things; e.g., it doesn't revert to the last public version when a new version is private. It does seem to satisfy the spirit of the issue, though. 

If we want "the last public version" to be exposed, that is more work. 

I suggest that we go with this fix for now.

**Comment by hyi** [comment]
@xuezhaokun @alvacouch @pkdash The PR looks good to me. +1 from me.

**Pull request merged**